Python/pystate.c

```diff
@@ -1732,6 +1759,13 @@ PyThreadState_Clear(PyThreadState *tstate)
 
     Py_CLEAR(tstate->context);
 
+    if (tstate->thread_is_exiting != NULL) {
+        _PyEventRc *erc = tstate->thread_is_exiting;
+        tstate->thread_is_exiting = NULL;
+        _PyEvent_Notify(&erc->event);
+        _PyEventRc_Decref(erc);
+    }
```

Comment:
Before gh-114839 we did almost exactly the same thing with `PyThreadState.on_delete`.  I guess we're coming back full circle. 😄